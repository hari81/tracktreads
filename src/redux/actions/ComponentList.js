import UCInspectionManager from '../../business/UCInspectionManager';

export const getComponentList = () => ({
	type: 'SELECT_COMPONENT_LIST',
});

export const getComponentListSuccess = data => ({
	type: 'FETCHING_COMPONENT_SUCCESS',
	data,
});

export const fetchComponentList = inspectionId => (dispatch) => {
	dispatch(getComponentList());
	UCInspectionManager.getComponentList(inspectionId)
		.then((response) => {
			dispatch(getComponentListSuccess(response));
		})
		.catch(err => console.log('err:', err));
};

export const selectComponent = item => ({
	type: 'SELECT_COMPONENT',
	data: item,
});

export const selectTool = tool => ({
	type: 'SELECT_COMPONENT_TOOL',
	data: tool,
});

export const updateComment = comment => ({
	type: 'UPDATE_INSPECTION_COMMENT',
	data: comment,
});

export const updateReading = (reading, worn) => ({
	type: 'UPDATE_INSPECTION_READING',
	reading,
	worn,
});

export const getTestPoint = value => ({
	type: 'SELECT_TEST_POINT',
	data: value,
});

export const getLocalToolImg = value => ({
	type: 'SELECT_LOCAL_TOOL_IMAGE',
	data: value,
});

export const getDealershipLimits = value => ({
	type: 'SELECT_DEALERSHIP_LIMITS',
	data: value,
});

export const savePhoto = photoPath => ({
	type: 'SAVE_COMPONENT_PHOTO',
	data: photoPath,
});

export const removePhoto = () => ({
	type: 'REMOVE_COMPONENT_PHOTO',
});

export const changesComponentList = (oldComponentList, selected) => ({
  type: 'CHANGES_COMPONENT_LIST',
  oldComponentList,
  selected
});

export const addRecommondsSave = (equnitAuto, objValue, selected) => ({
	type: 'ADD_RECOMMENDATIONS',
	data: objValue,
  equnit: equnitAuto,
  selected
});
