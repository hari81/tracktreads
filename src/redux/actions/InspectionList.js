import UCMainManager from '../../business/UCMainManager'

export const getUnsyncedList = () => {
    return {
        type: 'SELECT_UNSYNCED_LIST',
    }
}

export const getUnsyncedListSuccess = (data) => {
    return {
      type: 'FETCHING_DATA_SUCCESS',
      data: data,
    }
}

export const fetchUnsyncedList = () => {
    return (dispatch) => {
      dispatch(getUnsyncedList())
      UCMainManager.getUnsyncedEquipments()
        .then((response) => {
            dispatch(getUnsyncedListSuccess(response))
        })
        .catch((err) => console.log('err:', err))
    }
}

export const selectDeselectFromList = (inspectionId) => {
    return {
        type: 'SELECT_DESELECT_FROM_LIST',
        inspectionId: inspectionId
    }
}

export const pressRemove = () => {
    return {
        type: 'PRESS_REMOVE',
    }
}

export const selectCurrentInspection = (equipment) => {
    return {
        type: 'SELECT_CURRENT_INSPECTION',
        equipment: equipment
    }
}

export const updateEquipmentImg = (image) => {
    return {
        type: 'UPDATE_EQUIPMENT_IMAGE',
        data: image
    }
}
