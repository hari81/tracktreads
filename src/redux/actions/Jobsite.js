export const getJobsiteDetails = (value) => {
    return {
        type: 'GET_JOBSITE_DETAIL',
        value: value
    }
}

export const resetJobsiteDetails = () => {
    return {
        type: 'RESET_JOBSITE_DETAIL',
    }
}

export const updateImpact = (value) => {
    return {
        type: 'UPDATE_IMPACT',
        value: value
    }
}

export const updateAbrasive = (value) => {
    return {
        type: 'UPDATE_ABRASIVE',
        value: value
    }
}

export const updateMoisture = (value) => {
    return {
        type: 'UPDATE_MOISTURE',
        value: value
    }
}

export const updatePacking = (value) => {
    return {
        type: 'UPDATE_PACKING',
        value: value
    }
}

export const updateMeasure = (value) => {
    return {
        type: 'UPDATE_MEASURE',
        value: value
    }
}

export const updateLeftTrackSag = (value) => {
    return {
        type: 'UPDATE_LEFT_TRACK_SAG',
        value: value
    }
}

export const updateRightTrackSag = (value) => {
    return {
        type: 'UPDATE_RIGHT_TRACK_SAG',
        value: value
    }
}

export const updateLeftDryJoints = (value) => {
    return {
        type: 'UPDATE_LEFT_DRY_JOINTS',
        value: value
    }
}

export const updateRightDryJoints = (value) => {
    return {
        type: 'UPDATE_RIGHT_DRY_JOINTS',
        value: value
    }
}

export const updateLeftCannon = (value) => {
    return {
        type: 'UPDATE_LEFT_CANNON',
        value: value
    }
}

export const updateRightCannon = (value) => {
    return {
        type: 'UPDATE_RIGHT_CANNON',
        value: value
    }
}

export const updateLeftScallop = (value) => {
    return {
        type: 'UPDATE_LEFT_SCALLOP',
        value: value
    }
}

export const updateRightScallop = (value) => {
    return {
        type: 'UPDATE_RIGHT_SCALLOP',
        value: value
    }
}

export const updateInspectComment = (value) => {
    return {
        type: 'UPDATE_INSPECTION_COMMENT',
        value: value
    }
}

export const updateJobsiteComment = (value) => {
    return {
        type: 'UPDATE_JOBSITE_COMMENT',
        value: value
    }
}

export const savePhoto = (value) => {
    return {
        type: 'SAVE_PHOTO',
        value: value
    }
}

export const removePhoto = (photoType) => {
    return {
        type: 'REMOVE_PHOTO',
        photoType: photoType
    }
}
