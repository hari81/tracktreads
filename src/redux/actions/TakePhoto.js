export const openPhotoModal = (value) => {
    return {
        type: 'OPEN_PHOTO_MODAL',
        value: value
    }
}

export const closePhotoModal = () => {
    return {
        type: 'CLOSE_PHOTO_MODAL',
    }
}

export const takePhoto = (filePath) => {
    return {
        type: 'TAKE_PHOTO',
        path: filePath
    }
}

export const equipmentPhoto = (filePath) => {
  return {
    type: 'EQUIPMENT_PHOTO',
    filePath,
  };
}
export const updateComment = (text) => {
    return {
        type: 'UPDATE_PHOTO_COMMENT',
        value: text
    }
}