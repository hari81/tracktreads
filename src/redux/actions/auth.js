import UCIManager from '../../business/UCMainManager';
export const login = (username, password) => {
    return {
        type: 'LOGIN',
        username: username,
        password: password
    };
};
 
export const logout = () => {
    return {
        type: 'LOGOUT'
    };
};

export const loadActionType = () => {
    UCIManager.downloadTrackActionType();
};