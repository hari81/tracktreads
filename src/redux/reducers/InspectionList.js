import SQLiteManager from '../../database/SQLiteManager';
import UCMainManager from '../../business/UCMainManager';
// import { stat } from 'fs';

const defaultState = {

	// Get unsynced inspections
	dataFetched: false,
	isFetching: false,
	unsyncedList: [],

	// Select/ deselect items
	selectedList: [],

	// Current inspection
	currentInspection: {},
};

export default function reducer(state = defaultState, action) {
	// //////////////
	// FETCH DATA
	if (action.type === 'SELECT_UNSYNCED_LIST') {
		return {
			...state,
			isFetching: true,
			unsyncedList: [],
			selectedList: [],
			currentInspection: {},
		};
	}
	if (action.type === 'FETCHING_DATA_SUCCESS') {
		return {
			...state,
			isFetching: false,
			unsyncedList: action.data,
		};

		// //////////////////////////
		// SELECT/ DESELECT DATA
	}
	if (action.type === 'SELECT_DESELECT_FROM_LIST') {
		let newSelected = [];
		newSelected = newSelected.concat(state.selectedList);
		if (newSelected.includes(action.inspectionId)) {
			newSelected.forEach((element, index) => {
				if (element === action.inspectionId) {
					newSelected.splice(index, 1);
				}
			});
		} else {
			newSelected.push(action.inspectionId);
		}

		return {
			...state,
			selectedList: newSelected,
		};
	}
	if (action.type === 'UPDATE_EQUIPMENT_IMAGE') {
		return {
			...state,
			currentInspection: {
				...state.currentInspection,
				image: action.data,
			},
		};
  }
  // //////////////////////////
		// PRESS REMOVE
	if (action.type === 'PRESS_REMOVE') {
		let inspectionIds = state.selectedList.join(',');
		inspectionIds = `(${  inspectionIds  })`;
		SQLiteManager.deleteInspections(inspectionIds);

		return {
			...state,
			selectedList: [],
		};
	}
	if (action.type === 'SELECT_CURRENT_INSPECTION') {
		return {
			...state,
			currentInspection: action.equipment,
		};
  }
  if(action.type === 'EQUIPMENT_PHOTO') {
    console.log(state);
    return {
      ...state,
      currentInspection: { ...state.currentInspection, image: action.filePath },
    };
  }
	return state;
}
