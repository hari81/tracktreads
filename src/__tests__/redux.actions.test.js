import configureStore from 'redux-mock-store'

import * as authActions from '../redux/actions/auth'
import * as inspectionActions from '../redux/actions/InspectionList'

const middlewares = []
const mockStore = configureStore( middlewares )

const initialState = {}
const store = mockStore( initialState )

beforeEach( () => {
  store.clearActions();
} );

afterEach( () => {
  expect( store.getActions() ).toMatchSnapshot();
} )

////////////////
// LOGIN TEST

// Test by value
test('Dispatches the correct action and payload', () => {    
    store.dispatch(authActions.login('tomv', '123456'))    
    const expectedAction = [{
        "type": "LOGIN",
        "username": "tomv",
        "password": "123456",
    }]; 
    expect(store.getActions()).toEqual(expectedAction)
})

// Test by snapshot
test('Dispatches the correct action and payload', () => {    
    store.dispatch(authActions.login('tomv', '123456'))    
    expect(store.getActions()).toMatchSnapshot()

})

/////////////////////////////////
// SELECT/ DESELECT INSPECTION
test('Dispatches the correct action and payload', () => {    
    store.dispatch(inspectionActions.selectDeselectFromList(1))
    const expectedAction = [{
        type: 'SELECT_DESELECT_FROM_LIST',
        inspectionId: 1
    }]; 
    expect(store.getActions()).toEqual(expectedAction)
})
