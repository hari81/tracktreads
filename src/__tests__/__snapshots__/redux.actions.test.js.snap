// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "password": "123456",
    "type": "LOGIN",
    "username": "tomv",
  },
]
`;

exports[`Dispatches the correct action and payload 2`] = `
Array [
  Object {
    "password": "123456",
    "type": "LOGIN",
    "username": "tomv",
  },
]
`;

exports[`Dispatches the correct action and payload 3`] = `
Array [
  Object {
    "password": "123456",
    "type": "LOGIN",
    "username": "tomv",
  },
]
`;

exports[`Dispatches the correct action and payload 4`] = `
Array [
  Object {
    "inspectionId": 1,
    "type": "SELECT_DESELECT_FROM_LIST",
  },
]
`;
